File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        maxLength = 0
        left, right = 0, 0

        hash_set = set()

        while(right < len(s)):
            if s[right] in hash_set:
                hash_set.remove(s[left])
                left += 1    
            else:
                hash_set.add(s[right])
                right += 1

            maxLength = max(maxLength, right - left)    

        return maxLength


        
        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> set = new HashSet<>();
        int maxL = 0;

        int left = 0;
        int right = 0;

        while(right < s.length()) {
            char curr = s.charAt(right);
            if (!set.contains(s.charAt(right))) {
                set.add(curr);
                right++;
            } else {
                set.remove(s.charAt(left));
                left++;
            }

            maxL = Math.max(maxL, set.size());
        }

        return maxL;
        
    }
}